conans/client/graph/graph_binaries.py

```diff
@@ -173,24 +173,41 @@ def _find_existing_compatible_binaries(self, node, compatibles, remotes, update)
         node.binary = original_binary
         node._package_id = original_package_id
 
+    def _find_build_compatible_binary(self, node, compatibles):
+        original_binary = node.binary
+        original_package_id = node.package_id
+        for pkg_id, compatible in compatibles.items():
+            if not compatible.cant_build:
+                node._package_id = pkg_id  # Modifying package id under the hood, FIXME
+                self._compatible_found(node.conanfile, pkg_id, compatible)
+                node.binary = BINARY_BUILD
+                return
```

Comment:
This means that we get to build the first one provided by the compatibility plugin, as that comes before the compatibility method? Is there a way to output some info into what changed settings are going into the compilation? Does `compatible` here not store the generated output from the `compatibility` calls somewhere?